### PR TITLE
User, Review Models

### PR DESCRIPTION
**Pull request by oliviaoddo**

### CODE REVIEW

File: db/models/address.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const { STRING, INTEGER } = require('sequelize');
+
+module.exports = db => db.define('addresses', {
+  street: {
+    type: STRING,
+    allowNull: false,
+    validate: {
+      isAlphanumeric: true
```
Comment (oliviaoddo) [inline db/models/address.js]: Make sure this allows - .

File: db/models/address.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  },
+  city: {
+    type: STRING,
+    allowNull: false,
+    validate: {
+      isAlphanumeric: true
+    }
+  },
+  zip: {
+    type: INTEGER,
```
Comment (oliviaoddo) [inline db/models/address.js]: Store as string?

File: db/models/address.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    }
+  },
+  state: {
+    type: STRING,
+    allowNull: false,
+    validate: {
+      isAlphanumeric: true
+    }
+  },
+  country: {
```
Comment (oliviaoddo) [inline db/models/address.js]: default country to USA?

File: db/models/address.js (hunk 2 above)
Comment (oliviaoddo) [inline db/models/address.js]: Store as string?

File: db/models/review.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const { STRING, ENUM, TEXT } = require('sequelize');
+
+module.exports = db => db.define('reviews', {
+  title: STRING,
```
Comment (oliviaoddo) [inline db/models/review.js]: maybe add a max length? allow null false

File: db/models/address.js (hunk 1 above)
Comment (oliviaoddo) [inline db/models/address.js]: Make sure this allows - .

File: db/models/review.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  description: TEXT,
```
Comment (oliviaoddo) [inline db/models/review.js]: allowNull false?

File: db/models/user.js [hunk 6]
```diff
@@ -1,30 +1,42 @@
...
+  firstName: STRING,
+  lastName: STRING,
   email: {
     type: STRING,
     validate: {
       isEmail: true,
       notEmpty: true,
     }
   },
+  isAdmin: BOOLEAN
```
Comment (oliviaoddo) [inline db/models/user.js]: default value false

File: db/models/user.js [hunk 7]
```diff
@@ -1,30 +1,42 @@
...
+    attributes: {
+      exclude: ['password_digest']
+    }
+  },
+  getterMethods: {
+    fullName() {
+      return this.firstName + ' ' + this.lastName;
+    }
   },
   instanceMethods: {
```
Comment (oliviaoddo) [inline db/models/user.js]: why is this empty?

File: db/models/user.js [hunk 8]
```diff
@@ -34,9 +46,12 @@ module.exports = db => db.define('users', {
   }
 })
 
-module.exports.associations = (User, {OAuth, Thing, Favorite}) => {
+// USER ASSOCIATION IS LOCATED IN THE ADDRESS TABLE, but I'll make one here just in case
+module.exports.associations = (User, {
+  OAuth,
+  Address
+}) => {
   User.hasOne(OAuth)
```
Comment (oliviaoddo) [inline db/models/user.js]: User.hasMany(Review)

File: db/models/user.js [hunk 9]
```diff
@@ -34,9 +46,12 @@ module.exports = db => db.define('users', {
   }
 })
 
-module.exports.associations = (User, {OAuth, Thing, Favorite}) => {
+// USER ASSOCIATION IS LOCATED IN THE ADDRESS TABLE, but I'll make one here just in case
+module.exports.associations = (User, {
+  OAuth,
+  Address
```
Comment (oliviaoddo) [inline db/models/user.js]: User.hasOne(Address)

File: db/models/address.js [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    type: STRING,
+    allowNull: false
+  }
+})
+
+module.exports.associations = (Address, {
+  User
+}) => {
+  Address.hasMany(User)
```
Comment (oliviaoddo) [inline db/models/address.js]: Address.belongsToMany(User)

File: db/models/user.js (hunk 7 above)
Comment (stanleytiu7) [inline db/models/user.js:44]: I dunno what 'this' is

File: db/models/address.js (hunk 1 above)
Comment (stanleytiu7) [reply db/models/address.js]: I looked online for a regexp for the address validation and viewed a few stackoverflows. I found plenty of valid regexps, but they don't catch all addresses and have plenty of false-positives. One comment suggested not having address validation, so I think that that's a good idea for now.

**Pull request merged**